### PR TITLE
Add DEM layer with file-oscars type

### PR DESCRIPTION
**Pull request by JeroenVerstraelen**
* Added COP_DEM_GLO_30M_COG layer to layercatalog.json
* Added the 'file-oscars' layer type which uses the Sentinel2PyramidFactory 
* Currently the file-oscars layer uses a static cell-size of (30,30) but this will later be pulled out of the cube:dimensions field based on the step size and reference system.

### CODE REVIEW

File: layercatalog.json [hunk 1]
```diff
@@ -1272,5 +1272,56 @@
         "native_crs": "EPSG:4326"
       }
     }
+  },
+  {
+    "id": "COP_DEM_GLO_30M_COG",
```
Comment (jdries) [inline layercatalog.json:1277]: We'll need a better name for openEO.
Can we use 'COPERNICUS_30'? That aligns it with the Sentinelhub name. Details like 'COG' are not relevant anymore when using openEO.

File: layercatalog.json [hunk 2]
```diff
@@ -1272,5 +1272,56 @@
...
+      "spatial": {"bbox": [[-180, -90, 180, 90]]},
+      "temporal": {"interval": [["2010-12-12", null]]}
+    },
+    "cube:dimensions": {
+      "x": {"type": "spatial", "axis": "x", "step": 0.002777777777777778, "reference_system": "EPSG:4326"},
+      "y": {"type": "spatial", "axis": "y", "step": 0.002777777777777778, "reference_system": "EPSG:4326"},
+      "t": {"type": "temporal"},
+      "bands": {
+        "type": "bands",
+        "values": ["Copernicus_DSM_COG_10"]
```
Comment (jdries) [inline layercatalog.json:1309]: Let's use 'DEM' as band name, and use the aliases system to ensure that oscars gets the correct link title.

Comment (jdries) [reply layercatalog.json:1309]: Actually the complex link title in oscars is also questionable :-)

File: layercatalog.json (hunk 1 above)
Comment (JeroenVerstraelen) [reply layercatalog.json:1277]: Done.

File: layercatalog.json (hunk 2 above)
Comment (JeroenVerstraelen) [reply layercatalog.json:1309]: Done.

**Comment by JeroenVerstraelen** [comment]
The pull request does not appear to update with using my latest commits. Even though the commits are clearly shown at:
[https://github.com/Open-EO/openeo-geopyspark-driver/commits/feature/GDD-1273-copernicus-dem-openeo-layer](https://github.com/Open-EO/openeo-geopyspark-driver/commits/feature/GDD-1273-copernicus-dem-openeo-layer)

**Pull request merged**